pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -382,6 +382,10 @@ func (p *PriorityQueue) runPreEnqueuePlugin(ctx context.Context, pl framework.Pr
 func (p *PriorityQueue) addToActiveQ(pInfo *framework.QueuedPodInfo) (bool, error) {
 	pInfo.Gated = !p.runPreEnqueuePlugins(context.Background(), pInfo)
 	if pInfo.Gated {
+		// Unset pod info timestamp to the initial attempt timestamp,to exclude the preEnqueue failure time, pod will be in the back of the queue.
+		if pInfo.InitialAttemptTimestamp != nil {
+			pInfo.Timestamp = *pInfo.InitialAttemptTimestamp
+		}
```

Comment:
Remove this and add code above L393 b/c we just need to only set InitialAttemptTimestamp/Timestamp once - which is right after the pod is let go for the first time.

```go
		if pInfo.InitialAttemptTimestamp == nil {
		    now := time.Now()
			pInfo.InitialAttemptTimestamp = &now
			pInfo.Timestamp = now
		}
```